File: is_palindrome.py

def is_palindrome(string):
	length = len(string)
	string = string.lower()
	is_odd = (length % 2 == 1)
	ht = dict()
	for s in string:
		ht[s] = ht[s] + 1 if s in ht else 1
	has_visited_odd = not is_odd
	for key in ht:
		if ht[key] % 2 == 1:
			if has_visited_odd:
				return False
			else:
				has_visited_odd = True
	return True



def main():
	string = "TactCoa"
	isp = is_palindrome(string)
	print(isp)

if __name__ == "__main__":
	main()

File: is_palindrome.cpp
#include <iostream>
#include <string>
#include <map>

using namespace std;

bool isPalindromeHT(string str){

	int len = str.length();
	for(int i=0; i<len; i++){
		str[i] = tolower(str[i]);
	}
	int has_visited_odd = 0;
	
	map<char, int> ht;
	char* cstr= (char*) str.c_str();
	for(int i = 0; i<len; i++) {
		ht[cstr[i]] = ht[cstr[i]] + 1;
	}

	for(int i=0; i<ht.size(); i++){
		if((ht[i]%2)==1){
			if(has_visited_odd==1){
				return false;
			}
			else{
				has_visited_odd=1;
			}
		}
	}
	return true;
}

int main(){

	string str = "TactCoa";
	cout << isPalindromeHT(str) << endl; 

	return 0;
}
